gui/src/installer/step/mod.rs

```diff
@@ -111,6 +336,141 @@ fn bitcoind_default_address(network: &Network) -> String {
+            return Command::perform(async {}, |_| Message::Next);
+        };
+        Command::none()
+    }
+
+    fn apply(&mut self, ctx: &mut Context) -> bool {
+        if self.use_internal {
+            if ctx.internal_bitcoind_config.is_none() {
+                ctx.bitcoind_config = None; // Ensures internal bitcoind can be restarted in case user has switched selection.
+                let mut conf = match InternalBitcoindConfig::from_file(
+                    &internal_bitcoind_config_path(&ctx.data_dir),
+                ) {
+                    Ok(conf) => conf,
+                    Err(InternalBitcoindConfigError::FileNotFound) => InternalBitcoindConfig::new(),
+                    Err(e) => {
+                        self.error = Some(e.to_string());
+                        return false;
+                    }
+                };
+                // Insert entry for network if not present.
+                if conf.networks.get(&ctx.bitcoin_config.network).is_none() {
+                    let network_conf = match (get_available_port(), get_available_port()) {
+                        (Ok(rpc_port), Ok(p2p_port)) => {
+                            // In case ports are the same, user will need to click button again for another attempt.
+                            if rpc_port == p2p_port {
+                                self.error = Some(
+                                    "Could not get distinct ports. Please try again.".to_string(),
+                                );
+                                return false;
+                            }
+                            InternalBitcoindNetworkConfig {
+                                rpc_port,
+                                p2p_port,
+                                prune: PRUNE_DEFAULT,
+                            }
+                        }
+                        (Ok(_), Err(e)) | (Err(e), Ok(_)) => {
+                            self.error = Some(format!("Could not get available port: {}.", e));
+                            return false;
+                        }
+                        (Err(e1), Err(e2)) => {
+                            self.error =
+                                Some(format!("Could not get available ports: {}; {}.", e1, e2));
+                            return false;
+                        }
+                    };
+                    conf.networks
+                        .insert(ctx.bitcoin_config.network, network_conf);
+                }
+                ctx.internal_bitcoind_config = Some(conf.clone());
+                if let Err(e) = conf.to_file(&internal_bitcoind_config_path(&ctx.data_dir)) {
+                    self.error = Some(e.to_string());
+                    return false;
+                };
+            }
+        } else {
+            // In case the internal option was selected initially and then user returned to this page, stop bitcoind if it was started.
+            if ctx.internal_bitcoind_config.is_some() {
```

Comment:
It should be done in a `reload` method that you can add to the Step trait and that is triggered in the `previous` method of the installer,

Reply:
Or maybe even better would be to modify the view so when clicking on previous button, and other message is send to stop the bitcoind daemon then propagate the previous message.